### PR TITLE
ML-040/add sidekiq

### PR DESCRIPTION
**Pull request by KarinaKlevzhits**
# Description

Add sidekiq and sidekiq-cron gems
Config gems
Add workers(for reviewer-every monday, for redactor-every day)
Add specs for reviewer and redactor mailers
Add views for mailers
Add service

## Trello card link

https://trello.com/c/SB9dCO7j/53-ml-040-add-sidekiq

# Checklist:

- [x] My code fixes the problem or adds required functionality
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have added tests that prove that my feature works
- [x] New and existing unit tests pass locally with my changes

# Additional requirements:

- [ ] Does your code require any migrations to be ran?
- [x] Does your code require installing any new gems or system dependencies?
- [ ] Does your code require any changes to current deployment process?

# Additional Information

### CODE REVIEW

**Review by WhatWasThatt — DISMISSED** [review]
Really good job!

File: app/service/reviewer_notification_service.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class ReviewerNotificationService
+  def call
+    movies = Movie.where(release_date: (Time.zone.today..Time.zone.today + 7)).pluck(:name, :release_date)
+    User.joins(:role).where(roles: { name: 'reviewer' }).find_each do |user|
+      reviewer = user
+      call_mailer(movies, reviewer)
+    end
```
Comment (dmitrymalllpr) [inline app/service/reviewer_notification_service.rb:7]: call_mailer(movies,user)

File: app/mailers/redactor_mailer.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class RedactorMailer < ApplicationMailer
+  def redactor_notification(ratings, redactor)
+    @ratings = ratings
+    @redactor = redactor
+    mail(to: @redactor.email, subject: 'Redactor Notification')
+  end
+end
```
Comment (dmitrymalllpr) [inline app/mailers/redactor_mailer.rb:7]: remove repeated code please

**Review by dmitrymalllpr — CHANGES_REQUESTED** [review]

File: app/service/redactor_notification_service.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    User.joins(:role).where(roles: { name: 'redactor' }).find_each do |user|
+      redactor = user
+      call_mailer(ratings, redactor)
+    end
+  end
+
+  private
+
+  def call_mailer(ratings, redactor)
+    RedactorMailer.redactor_notification(ratings, redactor).deliver_now
```
Comment (twill11) [inline app/service/redactor_notification_service.rb:13]: dont use deliver or deliver_now please improve, https://medium.com/@coorasse/the-correct-emails-configuration-in-rails-c1d8418c0bfd

File: app/views/redactor_mailer/redactor_notification.html.erb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<h1>Hello,dear redactor  - <%= @redactor.name %>!</h1>
+<h3>New movie ratings for yesterday</h3> 
+ <p>
+   <% @ratings.each do |rating| %>
```
Comment (ShrewHools) [inline app/views/redactor_mailer/redactor_notification.html.erb:4]: maybe move each above tag 'p'?

File: app/views/redactor_mailer/redactor_notification.html.erb [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+    Rating: <%= rating[:rating] %> Movie: <%= rating[:"movies.name"]%> User: <%= rating[:"users.name"] %> </br>
```
Comment (ShrewHools) [inline app/views/redactor_mailer/redactor_notification.html.erb:5]: hash[:'String']
why?

File: app/service/reviewer_notification_service.rb [hunk 6]
```diff
@@ -0,0 +1,17 @@
+class ReviewerNotificationService
```
Comment (ShrewHools) [inline app/service/reviewer_notification_service.rb:1]: what is the meaning of class_service.rb file name in services folder?
i mean **_service**

File: app/service/redactor_notification_service.rb [hunk 7]
```diff
@@ -0,0 +1,17 @@
+class RedactorNotificationService
+  def call
+    ratings = Rating.where(created_at: Date.yesterday).joins(:movie, :user).pluck('rating', 'movies.name', 'users.name')
```
Comment (ShrewHools) [inline app/service/redactor_notification_service.rb:3]: can we use **select** and remove **.map {}**?

File: app/service/redactor_notification_service.rb [hunk 8]
```diff
@@ -0,0 +1,17 @@
+class RedactorNotificationService
```
Comment (ShrewHools) [inline app/service/redactor_notification_service.rb:1]: maybe you can put all the notifications in a folder? to make it easier for other users to use the services folder?

File: config/environments/development.rb [hunk 9]
```diff
@@ -62,4 +62,6 @@
 
   # Uncomment if you wish to allow Action Cable access from any origin.
   # config.action_cable.disable_request_forgery_protection = true
+  config.action_mailer.default_url_options = { host: "localhost:3000" } 
+  config.active_job.queue_adapter = :sidekiq
```
Comment (ShrewHools) [inline config/environments/development.rb:66]: do you really need this line?
will something break if you remove it?

**Review by WhatWasThatt — APPROVED** [review]

**Pull request merged**